code/modules/antagonists/eldritch_cult/eldritch_knowledge.dm

```diff
@@ -0,0 +1,249 @@
+  */
+/datum/eldritch_knowledge/proc/on_gain(mob/user)
+	to_chat(user, "<span class='warning'>[gain_text]</span>")
+	return
+/**
+  * What happens when you loose this
+  *
+  * This proc is called whenever antagonist looses his antag datum, put cleanup code in here
+  */
+/datum/eldritch_knowledge/proc/on_lose(mob/user)
+	return
+/**
+  * What happens every tick
+  *
+  * This proc is called on SSprocess in eldritch cultist antag datum. SSprocess happens roughly every second
+  */
+/datum/eldritch_knowledge/proc/on_life(mob/user)
+	return
+
+/**
+  * Special check for recipes
+  *
+  * If you are adding a more complex summoning or something that requires a special check that parses through all the atoms in an area override this.
+  */
+/datum/eldritch_knowledge/proc/recipe_snowflake_check(list/atoms,loc)
+	return TRUE
+
+/**
+  * What happens once the recipe is succesfully finished
+  *
+  * By default this proc creates atoms from result_atoms list. Override this is you want something else to happen.
+  */
+/datum/eldritch_knowledge/proc/on_finished_recipe(mob/living/user,list/atoms,loc)
+	if(result_atoms.len == 0)
+		return FALSE
+
+	for(var/X in result_atoms)
+		var/atom/A = X
+		new A(loc)
+
+	return TRUE
+
+/**
+  * Used atom cleanup
+  *
+  * Overide this proc if you dont want ALL ATOMS to be destroyed. useful in many situations.
+  */
+/datum/eldritch_knowledge/proc/cleanup_atoms(list/atoms)
+	for(var/X in atoms)
+		var/atom/A = X
+		if(!isliving(A))
+			qdel(A)
+	return
+
+/**
+  * Mansus grasp act
+  *
+  * Gives addtional effects to mansus grasp spell
+  */
+/datum/eldritch_knowledge/proc/on_mansus_grasp(atom/target, mob/user, proximity_flag, click_parameters)
+	return
+
+
+/**
+  * Sickly blade act
+  *
+  * Gives addtional effects to sickly blade weapon
+  */
+/datum/eldritch_knowledge/proc/on_eldritch_blade(atom/target, mob/user, proximity_flag, click_parameters)
+	return
+
+//////////////
+///Subtypes///
+//////////////
+
+/datum/eldritch_knowledge/spell
+	var/obj/effect/proc_holder/spell/spell_to_add
+
+/datum/eldritch_knowledge/spell/on_gain(mob/user)
+	var/obj/effect/proc_holder/S = new spell_to_add
+	user.mind.AddSpell(S)
+	return ..()
+
+/datum/eldritch_knowledge/spell/on_lose(mob/user)
+	user.mind.RemoveSpell(spell_to_add)
+	return ..()
+
+/datum/eldritch_knowledge/curse
+	var/timer = 5 MINUTES
+	var/list/fingerprints = list()
+
+/datum/eldritch_knowledge/curse/recipe_snowflake_check(list/atoms, loc)
+	fingerprints = list()
+	for(var/X in atoms)
+		var/atom/A = X
+		fingerprints |= A.return_fingerprints()
+	listclearnulls(fingerprints)
+	if(fingerprints.len == 0)
+		return FALSE
+	return TRUE
+
+/datum/eldritch_knowledge/curse/on_finished_recipe(mob/living/user,list/atoms,loc)
+
+	var/list/compiled_list = list()
+
+	for(var/mob/living/carbon/human/H in GLOB.human_list)
+		if(fingerprints[md5(H.dna.uni_identity)])
+			compiled_list |= H.real_name
```

Comment:
Can there be two humans with the same UI?

Reply:
I'm 99% sure no